clients/src/main/java/org/apache/kafka/clients/InFlightRequests.java

```diff
@@ -117,24 +113,24 @@ public int count(String node) {
      * Return true if there is no in-flight request directed at the given node and false otherwise
      */
     public boolean isEmpty(String node) {
-        Deque<NetworkClient.InFlightRequest> queue = requests.get(node);
-        return queue == null || queue.isEmpty();
+        return count(node) == 0;
```

Comment:
Generally, replacing an `isEmpty` check with `count` is not a good idea as you can easily regress in performance. In this case, the deque implementation may have a cheap `size()`, but it's brittle and not something I think we should do.

Reply:
OK.